hardware_frame_context.go

```diff
@@ -0,0 +1,48 @@
+package astiav
+
+//#include <libavcodec/avcodec.h>
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+type HardwareFrameContext struct {
+	c *C.struct_AVBufferRef
+}
+
+func AllocHardwareFrameContext(hdc *HardwareDeviceContext) (*HardwareFrameContext, error) {
+	if hfc := C.av_hwframe_ctx_alloc(hdc.c); hfc != nil {
+		return &HardwareFrameContext{c: hfc}, nil
+	}
+	return nil, fmt.Errorf("failed to allocate hardware frame context")
+}
+
+func (hfc *HardwareFrameContext) SetWidth(width int) {
+	frameCtx := (*C.AVHWFramesContext)(unsafe.Pointer((hfc.c.data)))
```

Comment:
Can you create a `func (hfc *HardwareFrameContext) data() *C.AVHWFramesContext` and use it all below functions instead of repeating `frameCtx := (*C.AVHWFramesContext)(unsafe.Pointer((hfc.c.data)))`?

It would end up being `hfc.data().width = C.int(width)` for instance